recipes/opengv/all/conanfile.py

```diff
@@ -54,26 +48,14 @@ def validate(self):
         #FIXME disable this one CCI has more RAM available
         if self.settings.compiler == "gcc" and self.options.shared:
             raise ConanInvalidConfiguration("Shared builds not supported with gcc since CCI errors out due to excessive memory usage.")
+        if self.settings.compiler == "clang" and self.settings.compiler.version == 12:
+            raise ConanInvalidConfiguration("Clang 12 builds fail on Conan CI.")
+
+    def layout(self):
+        cmake_layout(self)
```

Comment:
```suggestion
        cmake_layout(self, src_folder="src")
```
Avoid mixing source files with build cache, it's a mess in case you need to check something just after building

Reply:
I'm not sure if I understand this.
As is now, the sources will be extracted into `self.source_folder` and with your proposal we just move everything into a `src` subfolder, so `self.source_folder/src`. What's the benefit?

Reply:
not mixing conan cached files + project source files + cmake cached files. When developing a recipe, we usually take a look on build folder, to check missing artifacts for instance, but you put both obj files, libraries, mixing with the source folder, it's harder to check it.